### PR TITLE
fix memory adapter method to always return bool

### PR DESCRIPTION
**Pull request by maxp-hover**
`on?` will return nil if if the key is not present.

We want it to return false in that case.

### CODE REVIEW

File: lib/this_feature/adapters/memory.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def on?(flag_name, context: nil, data: {})
       def off?(flag_name, context: nil, data: {})
         on_result = on?(flag_name, context: context)
 
-        return if on_result.nil?
+        return true if on_result.nil?
```
Comment (matthewjf) [inline lib/this_feature/adapters/memory.rb:33]: Can we remove this line?

Comment (matthewjf) [reply lib/this_feature/adapters/memory.rb:33]: If `#on?` always returns a boolean, could off be
```ruby
def off?(flag_name, context: nil, data: {})
  !on?(flag_name, context: context, data: data)
end
```

**Review by matthewjf — APPROVED** [review]
👍

**Review by matthewjf — COMMENTED** [review]
wait missing `data`

File: lib/this_feature/adapters/memory.rb [hunk 2]
```diff
@@ -28,11 +28,7 @@ def on?(flag_name, context: nil, data: {})
       end
 
       def off?(flag_name, context: nil, data: {})
-        on_result = on?(flag_name, context: context)
-
-        return if on_result.nil?
-
-        !on_result
+        !on?(flag_name, context: context)
```
Comment (matthewjf) [inline lib/this_feature/adapters/memory.rb:31]:
```suggestion
        !on?(flag_name, context: context, data: data)
```

Comment (maxp-hover) [reply lib/this_feature/adapters/memory.rb]: Data isn't actually supported, but it doesn't hurt :) 
Good catch though!

**Pull request merged**